### PR TITLE
Enable reactor netty metrics

### PR DESCRIPTION
**Pull request by YenguiSeddik**
As mentioned here [spring-webflux-actuator-netty-thread-metrics](https://stackoverflow.com/questions/66028195/spring-webflux-actuator-netty-thread-metrics), the metrics exposed by react-netty are not enabled by default in spring boot. To enable them, we add a custom bean

### CODE REVIEW

File: src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+package org.gridsuite.gateway;
+
+import org.springframework.boot.web.embedded.netty.NettyServerCustomizer;
+import org.springframework.context.annotation.Configuration;
+import reactor.netty.http.server.HttpServer;
+
+import java.util.function.Function;
+
+// As discussed here https://stackoverflow.com/questions/66028195/spring-webflux-actuator-netty-thread-metrics,
```
Comment (antoinebhs) [inline src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java:16]: From what I read in this, I think we should add either a template for the URI or disable the metrics that we don't use.
```
	 * It is strongly recommended to provide template-like form for the URIs. Without a conversion to a template-like form,
	 * each distinct URI leads to the creation of a distinct tag, which takes a lot of memory for the metrics.
```
from their doc. 
See: https://quarkus.io/guides/telemetry-micrometer#define-tags

File: src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// the metrics expose by reactor-netty are not enabled by default in spring boot. To enable them we add the following bean
+@Configuration
+public class NettyMetricsConfiguration implements NettyServerCustomizer {
```
Comment (antoinebhs) [inline src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java:19]: Missing author?

File: src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java (hunk 1 above)
Comment (antoinebhs) [reply src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java:16]: you can see all the metrics it generates here: http://localhost:9000/actuator/prometheus 
if you navigate a bit in the app, it generates high cardinality metrics

Comment (antoinebhs) [reply src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java:16]: I would use something like:
https://gist.github.com/antoinebhs/8907cc2f9492ce84904d020b2d800977
to only store the metrics that we really use see documentation here: https://docs.micrometer.io/micrometer/reference/concepts/meter-filters.html
It works fine on my deployment, it only adds the two metrics (that are counter so no high cardinality issue/need to template etc)

Comment (antoinebhs) [inline src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java:16]: remove the link, just explain

Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java]: done

File: src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java (hunk 2 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java:25]: doen

File: src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Bean
+    public MeterFilter meterFilter() {
+        return MeterFilter.denyUnless(id -> {
+            String name = id.getName();
+            // Allow all non-reactor metrics
+            if (!name.startsWith("reactor.netty")) {
+                return true;
+            }
+            // Allow only the specific reactor metrics that we use
```
Comment (antoinebhs) [inline src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java:39]:
```suggestion
            // Allow only the specific reactor metrics that we use. If additional metrics are added, ensure the URIs are provided in a template-like format. 
            // Without this, each unique URI generates a separate tag, which takes a lot of memory.
```

File: src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import reactor.netty.http.server.HttpServer;
+
+import java.util.function.Function;
+
+/**
+ * @author Seddik Yengui <seddik.yengui_externe at rte-france.com>
+ */
+
+// the metrics expose by reactor-netty are not enabled by default in spring boot.
+// To enable them we add the following bean to customise the Netty HttpServer.
```
Comment (antoinebhs) [inline src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java:23]: // Enable Netty metrics that are not enabled by default in Spring Boot.

**Review by antoinebhs — COMMENTED** [review]
Comments improvements?

File: src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java (hunk 3 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java]: done

File: src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java (hunk 4 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/gateway/NettyMetricsConfiguration.java]: done

**Review by antoinebhs — APPROVED** [review]

**Pull request merged**